### PR TITLE
Better fuzzy heuristics

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
Continuation of the https://github.com/rust-analyzer/rust-analyzer/pull/6922, mainly created for a test.

Turns out our current completions tests were sorting the completions by label, I had to remove that to test the order properly and update this order in a bunch of tests (ergo the changes)

### CODE REVIEW

**Comment by matklad** [comment]
>Turns out our current completions tests were sorting the completions by label, I had to remove that to test the order properly and update this order in a bunch of tests (ergo the changes)

This is because we don't really force order of completions right now, it's the client who does the sorting. One of the things we should do around now-wish is to change that, and to move completion ordering to the server.

File: crates/completion/src/completions/unqualified_path.rs [hunk 1]
```diff
@@ -165,21 +160,47 @@ fn fuzzy_completion(acc: &mut Completions, ctx: &CompletionContext) -> Option<()
     Some(())
 }
 
+fn compute_fuzzy_completion_order_key(proposed_mod_path: &ModPath, user_input: &str) -> usize {
+    mark::hit!(certain_fuzzy_order_test);
+    let proposed_import_name = match proposed_mod_path.segments.last() {
+        Some(name) => name.to_string().to_lowercase(),
+        None => return usize::MAX,
+    };
+    let user_input = user_input.to_lowercase();
```
Comment (matklad) [inline crates/completion/src/completions/unqualified_path.rs:169]: Microoptimization, but let's move this allocation out of the sort_by_cached_key loop.

**Review by matklad — COMMENTED** [review]
r=me

**Comment by SomeoneToIgnore** [comment]
bors r=matklad

**Pull request merged**